dotCMS/src/main/java/com/dotcms/contenttype/transform/field/ImplClassFieldTransformer.java

```diff
@@ -23,28 +25,39 @@ public ImplClassFieldTransformer(final List<Field> fields) {
 
   @Override
   public Field from() throws DotStateException {
-    if (this.genericFields.size() == 0)
-      throw new DotStateException("0 results");
-    Field field = impleClass(this.genericFields.get(0));
+    if (this.genericFields.size() == 0) {
+        throw new DotStateException("There are no fields to transform!");
+    }
+    final Field field = impleClass(this.genericFields.get(0));
     return field;
 
   }
 
   private static Field impleClass(final Field genericField) {
-    FieldBuilder builder = null;
+    FieldBuilder builder;
     try {
       builder = FieldBuilder.builder(genericField);
       return builder.build();
-    } catch (Exception e) {
-      throw new DotStateException(e.getMessage(), e);
+    } catch (final Exception e) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5458294)